MaxText/max_utils.py

```diff
@@ -351,3 +109,4 @@ def _cross_entropy_with_logits_bwd(
 
 cross_entropy_with_logits.defvjp(_cross_entropy_with_logits_fwd,
```

Comment:
this custom cross_entropy seems useful for many model whose final task is classification, so I believe it belongs in the shared file.